crates/typst/src/layout/inline/mod.rs

```diff
@@ -876,12 +880,15 @@ fn linebreak_optimized<'a>(
     }
 
     // Cost parameters.
-    const HYPH_COST: Cost = 0.5;
-    const RUNT_COST: Cost = 0.5;
+    const DEFAULT_HYPH_COST: Cost = 0.5;
+    const DEFAULT_RUNT_COST: Cost = 0.5;
     const CONSECUTIVE_DASH_COST: Cost = 0.3;
     const MAX_COST: Cost = 1_000_000.0;
     const MIN_RATIO: f64 = -1.0;
 
+    let hyph_cost = DEFAULT_HYPH_COST * p.costs.hyphenation.unwrap_or(Ratio::one()).get();
```

Comment:
how about adding methods to `Costs` that return the cost and have internally the default costs and apply the necessary unwrap_or stuff / produce bools for widow/orphan?